pychromecast/__init__.py

```diff
@@ -334,6 +364,68 @@ def join(self, timeout=None):
         """
         self.socket_client.join(timeout=timeout)
 
+    def start_dynamic_host_tracking(self):
+        """
+        Start dynamic host tracking, register for connection status events
+        """
+        self.socket_client.register_connection_listener(self)
+
+    def new_connection_status(self, status):
+        """
+        Callback for socket client connection status
+        """
+        if not status.status == CONNECTION_STATUS_DISCONNECTED:
+            return  # Not interesting
+
+        def discovery_timeout():
+            """
+            Internal scheduled timeout for discovery, 5 seconds
+            """
+            time.sleep(10)
+            if not self.browser:
+                return  # Change in ip detected, discovery did not time out
+
+            self.logger.info('Discovery timed out, stopping')
```

Comment:
You use `logger.info` a lot. I think that for example this one should be reduced to debug level.